src/Sentry.Unity/UnityEventProcessor.cs

```diff
@@ -108,7 +108,8 @@ private void PopulateDevice(Device device)
 
             // This is the approximate amount of system memory in megabytes.
             // This function is not supported on Windows Store Apps and will always return 0.
-            if (SystemInfo.systemMemorySize != 0)
+            // Additionally: iOS issue with reporting negative memory size https://issuetracker.unity3d.com/issues/ios-if-memory-of-device-is-2gb-over-systeminfo-dot-systemmemorysize-is-returned-wrong-value-on-ios
+            if (SystemInfo.systemMemorySize > 0)
             {
                 device.MemorySize = SystemInfo.systemMemorySize * 1048576L; // Sentry device mem is in Bytes
```

Comment:
I didn't measure so this might not make sense but `systemMemorySize` is an `extern` method so to reduce to cost of P/Invoke we could do this:

```suggestion
            if (SystemInfo.systemMemorySize is {} systemMemorySize && systemMemorySize > 0)
            {
                device.MemorySize = mem * 1048576L; // Sentry device mem is in Bytes
```

There's probably a pattern to check directly the value being `>` instead of `is {} &&`.

Reply:
JFYI, this value will be stored in main thread data aka cached in my PR.